### PR TITLE
Display: Add wrapped framebuffer and framebuffer emulation script

### PR DESCRIPTION
**Pull request by ada-phillips**
Wrap framebuffer reference to better allow file output for test
visualization
Move display config fields to their own section
Add python script for making a FIFO pipe and loading data from
said pipe to a pygame screen for visualization and debugging
Reverse endinan-ness of display output

### CODE REVIEW

File: default.yaml [hunk 1]
```diff
@@ -6,17 +6,24 @@
...
   default_up_speed: 3.4
   default_down_speed: 3.4
   default_wait_before_exposure: 2.2
   default_wait_after_exposure: 1.5
   pause_lift: 100
 
+# This section holds fields pertaining to the display used by the printer
+display:
+  frame_buffer: /dev/fb0
+  bit_depth:
```
Comment (PaulGD03) [inline default.yaml:20]: Maybe we can assume 565 as the default? Afaik most of these screen drivers use RGB565.

File: test/test.yaml [hunk 2]
```diff
@@ -1,14 +1,11 @@
-# This is the default Odyssey configuration file for the Prometheus MSLA. It is
-# meant to be paired with the latest Prometheus MSLA klipper config.
+# This is the Odyssey configuration file for local testing only. It is not intended
+# for real hardware, and does not carry particularly realistic values.
 
 # This section holds the config fields related to the printer, such as its serial
 # connection and its frame buffer specs
 printer:
-  serial: /dev/pts/12 # Mock port opened with `python3 test/mockSerial.py`
```
Comment (PaulGD03) [inline test/test.yaml:7]: Maybe we can come up with a way to automatically determine this by scanning for a mockSerial.py process?

File: test/test.yaml [hunk 3]
```diff
@@ -17,6 +14,17 @@ printer:
   default_wait_after_exposure: 1.5
   pause_lift: 100
 
+# This section holds fields pertaining to the display used by the printer
+display:
+  #frame_buffer: /tmp/emulatedFramebuffer # Use with fbEmulator.py
+  frame_buffer: /dev/null # Dump frames to null for testing
+  bit_depth:
```
Comment (PaulGD03) [inline test/test.yaml:21]: CC: Maybe we can assume 565 as the default? Afaik most of these screen drivers use RGB565.

File: test/fbEmulator.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+bytes_per_pixelgroup=int(sum(real_bit_depth)/8)
+
+fake_bit_depth=8
+
+
+#sliced_x, sliced_y = (6480,3600)
+#zoom_ratio=1
+
+sliced_x, sliced_y = (192,108)
```
Comment (PaulGD03) [inline test/fbEmulator.py:25]: Just as a question, I'm guessing this is referring to 1080p through a 10x upscaling?

**Review by PaulGD03 — COMMENTED** [review]
Looking very good overall!

File: default.yaml (hunk 1 above)
Comment (ada-phillips) [reply default.yaml:20]: I don't think the Prusa screens do? And regardless, I'd like to keep the option to change it if necessary

File: test/test.yaml (hunk 2 above)
Comment (ada-phillips) [reply test/test.yaml:7]: Yeaaah, this whole thing is getting a drastic revamp with the testing overhaul. This functionality will not be sticking around.

File: test/test.yaml (hunk 3 above)
Comment (ada-phillips) [reply test/test.yaml:21]: See above

File: test/fbEmulator.py (hunk 4 above)
Comment (ada-phillips) [reply test/fbEmulator.py:25]: Yeah, for ease of testing on a regular ol' monitor.

**Review by PaulGD03 — APPROVED** [review]

**Pull request merged**